slicerator.py

```diff
@@ -48,23 +60,134 @@ def __init__(self, ancestor, indices=None, length=None):
         >>> type(v3)
         generator
         """
-        if indices is None:
+        if indices is None and length is None:
             try:
-                indices = range(len(ancestor))
+                length = len(ancestor)
+                indices = range(length)
             except TypeError:
-                raise ValueError("The indices parameter is required in this "
+                raise ValueError("The length parameter is required in this "
                                  "case because len(ancestor) is not valid.")
-        if length is None:
+        elif indices is None:
+            indices = range(length)
+        elif length is None:
             try:
                 length = len(indices)
             except TypeError:
                 raise ValueError("The length parameter is required in this "
                                  "case because len(indices) is not valid.")
+
+        # when list of propagated attributes are given explicitly,
+        # take this list and ignore the class definition
+        if propagate_attrs is not None:
+            self._propagate_attrs = propagate_attrs
+        else:
+            # check propagated_attrs field from the ancestor definition
+            if hasattr(ancestor, 'propagate_attrs'):
+                self._propagate_attrs = ancestor.propagate_attrs
+            else:
+                self._propagate_attrs = []
+
+            # add methods having the _propagate hook
+            for name in dir(ancestor):
+                if hasattr(getattr(ancestor, name), '_propagate_hook'):
```

Comment:
this actually evaluates *all* properties of `ancestor`. Not desired behaviour, I think. Need a workaround here.